### DISCUSSION TITLE
[Feature] Deploy rustdesk client from command line.

### DISCUSSION THREAD

**Discussion by jayagami** [rustdesk/rustdesk#2145] [General]
Hi, thanks for your great work.
I wonder if it is possible to support rustdesk deployment from command line.

Suppose I have a remote linux workstation with ssh connections. 
Currently, if I want install rustdesk, I have to physically access the screen. Or by another remote control tools like vnc or rdp.

Perhaps, I can just install rustdesk from command line, start it by command or systemed and modify configuration from file.

```
❯ rustdesk

(rustdesk:2646870): Gtk-WARNING **: 21:41:54.153: cannot open display:

```

**Comment by KiLLeRRaT** [comment]
This would be great.  I have a system and only have SSH access.  It'd be great to SSH in, start rustdesk, and then resume my graphical session with everything that was open.

**Comment by brunetton** [comment]
I'd really need this ! I can login to my mother's machine with ssh, and I'd really need to be able to start a graphical control from inside the ssh connection on her computer; so by ssh. 
For now I'm using VNC, but video codecs are super slower: `x11vnc -localhost -display :0 -auth guess -quiet -forever` and I then access via SSH tunnel. This is quite complicated and super slow. I'd love to be able to do something like I can do with TeamViewer: 

```bash
rustdesk --set-password insecure
rustdesk --get-id
```

Or by editing config file

Thanks

**Reply by dinger1986** [reply]
you can do this https://rustdesk.com/docs/en/self-host/client-deployment/ https://rustdesk.com/docs/en/client/#command-line-parameters

**Reply by brunetton** [reply]
Indeed I finally found it reading tons of issues and cherry-picking command line arguments from there :astonished: 

To resume those scripts; the two main commands here are:

```bash
rustdesk --password mySecretPassword
sudo rustdesk --get-id
```

So my 2 cents would be: use a command line arguments parser ! Something like [StructOpt](https://docs.rs/structopt/latest/structopt/); so users could just do `rustdesk --help` and have access to an always up-to-date doc :+1: 

Thanks for your answer

**Reply by sylvorg** [reply]
Did you find a solution to opening RustDesk using ssh, though, or are you still encountering the `cannot open display` error?

**Comment by gheoan** [comment]
You need to set the `DISPLAY` env variable.

`DISPLAY=:0 rustdesk`

@sylvorg @brunetton @jayagami

**Reply by brunetton** [reply]
Thanks :)
